src/promptflow/promptflow/exceptions.py

```diff
@@ -245,37 +238,35 @@ def get_error_info(cls, e: Exception):
                 cls._error_detail(e),
             )
 
-        return ErrorCategory.UNKNOWN, cls._error_type(e), ErrorTarget.UNKNOWN, "", cls._error_detail(e)
-
-    @classmethod
-    def select_exception(cls, e: Exception):
-        """Select the exception  in e and e.__cause__, and prioritize the Exception defined in the promptflow."""
-
-        if isinstance(e, PromptflowException):
-            return e
-
-        # raise Exception("message") from PromptflowException("message")
-        if e.__cause__ and isinstance(e.__cause__, PromptflowException):
-            return e.__cause__
-
-        return e
+        return (
+            ErrorCategory.SYSTEM_ERROR,
+            cls._error_type(e),
+            cls._error_target(e),
+            cls._error_message(e),
+            cls._error_detail(e),
+        )
 
     @classmethod
-    def _is_system_error(cls, e: Exception):
+    def _is_system_error(cls, e: BaseException):
         if isinstance(e, SystemErrorException):
             return True
+        if isinstance(e, HttpResponseError):
+            status_code = str(e.status_code)
+            # Except for 429, 400-499 are all client errors.
+            if not status_code.startswith("4") and status_code != "429":
```

Comment:
Because we have encountered many 429 issues before, although it was due to client requests exceeding the rate limit, it was actually due to the server not releasing the link in a timely manner, resulting in insufficient links. Ultimately, it was necessary for the server to fix them.
![image](https://github.com/microsoft/promptflow/assets/23182548/ff0fbee7-e538-41df-83e4-b48a81ed1463)